lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -310,7 +310,13 @@ def _patch(self, _) -> None:
                     self._delete_and_create_service(client)
                 else:
                     self._create_lb_service(client)
-            client.patch(Service, self.service_name, self.service, patch_type=PatchType.MERGE)
+            client.patch(
+                Service,
+                self.service_name,
+                self.service,
+                patch_type=PatchType.APPLY,
```

Comment:
is there a reason why we'd want `APPLY` here but `STRATEGIC` (the default) in [istio-beacon](https://github.com/canonical/istio-beacon-k8s-operator/blob/c389ccc65d82d138197a222d20191b8e7e4b9317/src/charm.py#L362C35-L362C40)?

Reply:
I'm not sure this is an issue, but I'm raising it for discussion...

A consequence of using `APPLY` here is that lightkube will raise an exception if someone else already "owns" this object (ex: try doing two applies to an object using lightkube, each with a different field manager).  

This wouldn't be so bad if we owned the object, since we could be confident that nobody will set the field manager but us, but in this case Juju owns it.  So if Juju ever tries to apply to the object, we could have either of:
1. Juju sets field manager, so then our code will raise an exception here
2. our code sets field manager on apply, and if Juju ever tries to reconcile the object back to normal then Juju will hit an exception

Probably (1) isn't an issue (because we'd see it in integration tests), but (2) feels like an implementation detail that Juju owns.

Reply:
Yeah, you're correct, I was under the impression that juju won't own this as it's an additional LB that we create, but I'm realizing now that you can also request a `ClusterIP` service with this lib which will override a juju owned object.

Not sure if `Strategic` patch will give us what we want but will try it out.